lms/djangoapps/learner_home/serializers.py

```diff
@@ -284,26 +324,78 @@ class LearnerEnrollmentSerializer(serializers.Serializer):
     Info for displaying an enrollment on the learner dashboard.
     Derived from a CourseEnrollment with added context.
     """
+    requires_context = True
 
     course = CourseSerializer()
     courseRun = CourseRunSerializer(source="*")
     enrollment = EnrollmentSerializer(source="*")
     certificate = CertificateSerializer(source="*")
+    entitlements = serializers.SerializerMethodField()
 
     # TODO - remove "allow_null" as each of these are implemented, temp for testing.
     courseProvider = CourseProviderSerializer(allow_null=True)
     gradeData = GradeDataSerializer(allow_null=True)
-    entitlements = EntitlementSerializer(allow_null=True)
     programs = ProgramsSerializer(allow_null=True)
 
+    def get_entitlements(self, instance):
+        """
+        If this enrollment is the fulfillment of an entitlement, include information about the entitlement
+        """
+        entitlement = self.context['fulfilled_entitlements'].get(str(instance.course_id))
+        if entitlement:
+            return EntitlementSerializer(entitlement, context=self.context).data
+        else:
+            return {}
+
 
 class UnfulfilledEntitlementSerializer(serializers.Serializer):
-    """Serializer for an unfulfilled entitlement"""
+    """
+    Serializer for an unfulfilled entitlement.
+    This should have the same keys as the LearnerEnrollmentSerializer.
+    We are flattening the two lists into one "course card" list and so should be the same shape.
+    """
+
+    requires_context = True
+
+    ENROLLMENT_DATA = {
+        'accessExpirationDate': None,
+        'isAudit': False,
+        'hasStarted': False,
+        'hasFinished': False,
+        'isVerified': False,
+        'canUpgrade': False,
+        'isAuditAccessExpired': False,
+        'isEmailEnabled': False,
+        'hasOptedOutOfEmail': False,
+        'lastEnrolled': None,
+        'isEnrolled': False,
+    }
```

Comment:
If we had a more straightforward standard use case for these serializers, where we were just passing around models and pulling fields off of them, then yeah, it'd probably be best to have the EnrollmentSerializer have these as defaults and pass it some sort of dummy value, but we have so many serializermethodfields and use of context that I think this is better.

Plus rather than acting like we have enrollment data but having a serializer that allows some missing fields, I'd rather be explicit that hey, this is what we're retuning. There's no serialization. There's no data being pulled from anywhere. It's this constant, in this shape.

I will add additional comments and rename the variable